lib/CMakeLists.txt

```diff
@@ -69,6 +71,21 @@ endif(APPLE)
 include(GrMiscUtils)
 GR_LIBRARY_FOO(gnuradio-pulsed_power_daq)
 
+########################################################################
+# add opencmw
+########################################################################
+include(FetchContent)
+
+FetchContent_Declare(
+        opencmw-cpp
+        GIT_REPOSITORY https://github.com/fair-acc/opencmw-cpp.git
+        GIT_TAG main # todo: use proper release once available
+)
+
+set(ENABLE_TESTING OFF)
+FetchContent_MakeAvailable(opencmw-cpp)
+target_link_libraries(gnuradio-pulsed_power_daq majordomo)
+
```

Comment:
yes, we are discussing a similar concept to decouple the gnuradio part from the opencmw part. Additionally this is not meant to be merged into the main branch at the moment.

Reply:
Alright .. in case you decide to use the concept .. [here](https://gitlab.com/al.schwinn/DigitizerClass2/-/blob/master/src/DigitizerClass2/Common/Sink.cpp#L217) the related call on the fesa side. An object is passed as `userdata` and later statically cast back [inside the callback method](https://gitlab.com/al.schwinn/DigitizerClass2/-/blob/master/src/DigitizerClass2/Common/Sink.cpp#L52).